pycontrails/core/fleet.py

```diff
@@ -134,18 +132,19 @@ def from_seq(
 
         if copy:
             seq = [fl.copy() for fl in seq]
-        else:
+        elif not isinstance(seq, (list, tuple)):
             seq = list(seq)
 
         assert seq, "Cannot create Fleet from empty sequence."
+        fuel = seq[0].fuel
         for fl in seq:
-            cls._validate_fl(fl, seq[0].fuel, attrs, broadcast_numeric)
+            cls._validate_fl(fl, fuel, attrs, broadcast_numeric)
 
         # Surprisingly, it's faster to call np.concatenate on each variable
         # separately then it is to call pd.concat on the sequence of fl.dataframe.
         # We do this concatenation below.
         data = {var: np.concatenate([fl[var] for fl in seq]) for var in attrs["data_keys"]}
-        return cls(data=data, attrs=attrs, copy=False)
+        return cls(data=data, attrs=attrs, copy=False, fuel=fuel)
```

Comment:
I regret adding this as a class init param. Fuel should probably have been managed as an attr.

Reply:
I agree, although the `fuel` attribute isn't too much of a headache. We definitely could start using `attrs` as the source of truth for `fuel` and add a `DeprecationWarning` when `Flight.fuel` is accessed. Perhaps we could do this as part of a larger cleanup when we get some feedback from the community.